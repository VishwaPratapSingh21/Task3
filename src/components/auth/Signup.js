import React, {Component} from 'react';
import axiosInstance from '../Apicalls';
import {notify} from 'react-notify-toast'
import {Link, Redirect} from 'react-router-dom';
import {Nav} from './Landing'

class Signup extends Component {
    state = {
        password_field: '',
        email_field: '',
        username: '',
        error:null,
    }

    handleInputChange = (event) => {
        const {name, value} = event.target;
        this.setState({[name]: value});
    }

    handleRegister = (event) => {
        const {email_field: email, password_field: password, username: username} = this.state;

        event.preventDefault();
        axiosInstance
            .post('auth/register', {username, email, password})
            .then(response => {
                notify.show(response.data.message, 'success', 4000);
                this
                    .props
                    .history
                    .push('/login');
            })
            .catch(error => {
                if (error.response) {
                    this.setState({error: error.response.data.message})
                    // notify.show(error.response.data.message, 'error', 3000)
                } else if (error.request) {
                    notify.show("Request not made", 'error', 3000)
                }
            });
    }

    render() {
        const {username, password_field, email_field} = this.state;
        let user = this.state.username
        
        if(this.props.loggedIn){
            // if one is loggedin it should not show signup anymore
            return(<Redirect to='/dashboard'/>)
        }

        return (
            
            <div className="background">
                <div id="signbar">
                        <div >
                    <i className="fa fa-user-circle fa-5x" style={{color:"#26A69A"}}></i>
                    <div> REGISTER </div>
                </div>
                    <form onSubmit={this.handleRegister}>
                    {this.state.error ? (
                    <div className="alert alert-danger">{this.state.error}
                     <button
                    type="button"
                    class="close"
                    data-dismiss="alert"
                    aria-label="Close"
                  >
                    <span aria-hidden="true">&times;</span>
                  </button></div>
                  ) : (
                    ""
                  )}
                        <div className="input-group">
                            <span className="input-group-addon pr-4" id="username"><i className='fa fa-user'/></span>
                            <input
                                id="username"
                                name="username"
                                aria-describedby="username"
                                className="form-control"
                                placeholder="example"
                                onChange={this.handleInputChange}
                                value={username}/>
                        </div>
                        <br/>
                        <div className="input-group">
                            <span className="input-group-addon pr-4" id="email"><i className='fa fa-envelope'/></span>
                            <input
                                id="emailsignup"
                                name="email_field"
                                aria-describedby="email"
                                className="form-control"
                                placeholder="example@gmail.com"
                                onChange={this.handleInputChange}
                                value={email_field}/>
                        </div>
                        <br/>
                        <div className="input-group">
                            <span className="input-group-addon pr-4" id="email"><i className='fa fa-lock'/></span>
                            <input
                                type="password"
                                name="password_field"
                                className="form-control"
                                onChange={this.handleInputChange}
                                value={password_field}
                                placeholder="Password"/>
                        </div>
                        <br/>
                            <div className="text-center">
                                <input name="signup" id="signup" value="Signup" className="btn btn-default btn-lg" type="submit"/>
                            
                        </div>

                        <div className="change_link">
                            Already a member ?
                            <Link to="/login"><h4 className="toregister">Login here</h4>
                            </Link>
                        </div>

                    </form>
                </div>
            </div>
        );
    }
}
export default Signup;
