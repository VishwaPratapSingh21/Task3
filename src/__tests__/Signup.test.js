import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Signup from '../components/auth/Signup';

describe('Signup component', () => {
  const wrapper = shallow(<Signup />);
  const preventDefault = jest.fn();

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('has a form', () => {
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('form').simulate('submit', { preventDefault }));
    expect(preventDefault).toBeCalled();
  });

  it('renders inputs', () => {
    expect(wrapper.find('input').length).toBe(4);
  });

  it('renders component in div', () => {
    expect(wrapper.find('div').length).toBe(9);
  });

  it('renders component in span', () => {
    expect(wrapper.find('span').length).toBe(3);
  });

  it('has the correct form fields', () => {
    expect(wrapper.find('#emailsignup')).toHaveLength(1);
    expect(wrapper.find('[name="password_field"]')).toHaveLength(1);
    expect(wrapper.find('#signup')).toHaveLength(1);
  });

  it('has empty form fields', () => {
    expect(wrapper.find('#emailsignup').props().value).toEqual('');
    expect(wrapper.find('[name="password_field"]').props().value).toEqual('');
    expect(wrapper.find('#signup').props().value).toEqual('Signup');
  });

  it('Form fields update when state changes', () => {
    wrapper.setState({ password_field: 'hi', email_field: 'mail' });
    expect(wrapper.find('#emailsignup').props().value).toEqual('mail');
    expect(wrapper.find('[name="password_field"]').props().value).toEqual('hi');
  });
});
