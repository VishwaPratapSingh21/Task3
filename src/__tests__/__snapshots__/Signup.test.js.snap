// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component renders properly 1`] = `
<div
  className="background"
>
  <div
    id="signbar"
  >
    <div>
      <i
        className="fa fa-user-circle fa-5x"
        style={
          Object {
            "color": "#26A69A",
          }
        }
      />
      <div>
         REGISTER 
      </div>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="input-group"
      >
        <span
          className="input-group-addon pr-4"
          id="username"
        >
          <i
            className="fa fa-user"
          />
        </span>
        <input
          aria-describedby="username"
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="example"
          value=""
        />
      </div>
      <br />
      <div
        className="input-group"
      >
        <span
          className="input-group-addon pr-4"
          id="email"
        >
          <i
            className="fa fa-envelope"
          />
        </span>
        <input
          aria-describedby="email"
          className="form-control"
          id="emailsignup"
          name="email_field"
          onChange={[Function]}
          placeholder="example@gmail.com"
          value=""
        />
      </div>
      <br />
      <div
        className="input-group"
      >
        <span
          className="input-group-addon pr-4"
          id="email"
        >
          <i
            className="fa fa-lock"
          />
        </span>
        <input
          className="form-control"
          name="password_field"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <br />
      <div
        className="text-center"
      >
        <input
          className="btn btn-default btn-lg"
          id="signup"
          name="signup"
          type="submit"
          value="Signup"
        />
      </div>
      <div
        className="change_link"
      >
        Already a member ?
        <Link
          replace={false}
          to="/login"
        >
          <h4
            className="toregister"
          >
            Login here
          </h4>
        </Link>
      </div>
    </form>
  </div>
</div>
`;
